sympy/core/numbers.py

```diff
@@ -2956,31 +2956,21 @@ def __le__(self, other):
             other = _sympify(other)
         except SympifyError:
             raise TypeError("Invalid comparison %s <= %s" % (self, other))
-        if other.is_extended_real:
-            if other is S.Infinity:
-                return S.true
-            elif other.is_finite or other is S.NegativeInfinity:
-                return S.false
-            elif other.is_extended_nonpositive:
-                return S.false
-            elif other.is_infinite and other.is_extended_positive:
-                return S.true
+        if other.is_infinite and other.is_extended_positive:
+            return S.true
+        elif other.is_real or other.is_extended_nonpositive:
```

Comment:
This can be further reduced to `other.is_extended_real`.

Reply:
This is `oo <= other`. If I have `other = Symbol('x', extended_real=True)` then other could possibly be oo and `oo <= oo` evaluates to True.

Reply:
That case is handled in the `if` part. I am suggesting for the `elif` part.

Reply:
I get:
```
is_real -> a=(-oo,oo)
extended_nonpositive -> b=[-oo, 1)
U(a,b)->[-oo,oo)
```
but `extended_real -> [-oo,oo]`, which is not the same. I think the result is 
```
other is not S.Infinity and other.is_extended_real
```

Reply:
If other is oo then it will be handled in the if. If other is `Symbol('x', extended_real=True)` it should pass through both cases and the result should be unevaluated because `x` *might* be oo but it also might not:
```
>>> from sympy import *
>>> x = Symbol('x', extended_real=True)
>>> oo <= x
oo <= x
>>> oo <= oo
True
```

Reply:
We can't use `other is not S.Infinity` because it won't exclude a symbol with the same assumptions i.e. `Symbol('y', extended_positive=True, infinite=True)` or a symbol that is *possibly* infinite e.g. `Symbol('x', extended_real=True)`.